### ISSUE TITLE
RAZ Reset?

### ISSUE THREAD

**Issue by b3ck** [ginbot86/ColorLCDVape-RE#3]
I wish there was a way to reset the e-liquid meter/counter, for people that refill it, that would come in handy, like a simple program that you can run after plugging it in via USB Type-C would be perfect.

**Comment by ginbot86** [comment]
The requirement for SWD hardware will continue to be a challenge, but another researcher did come up with some software tools to help with editing the juice level: https://github.com/xbenkozx/RAZ-RE/tree/main/Flash%2Freflasher

**Comment by b3ck** [comment]
So I have the "RAZ Vape DC25000 Disposable", what can I do to help? Is there a way to extract the firmware from this? then make modifications etc..?

**Comment by ginbot86** [comment]
The firmware extraction would be with the aforementioned SWD method (I just used my Segger J-Link with the J-Mem tool to grab 0x10000 (64k) bytes from address 0x08000000. The graphics from SPI Flash were dumped by desoldering the chip and mounting it to a 0.5mm-pitch SMD breakout board.

I noticed there are two different "BOOST Mode" graphics in the Kraze model I dumped: one is a plain large flame symbol, and an unused one that looks like tiger's faces in the flames. Which one does yours have in normal operation?

![hdmega-boost-alt](https://github.com/user-attachments/assets/b4b2e9bd-0630-4d68-a9c7-bc4cde448a2f)
![hdmega-boost](https://github.com/user-attachments/assets/9fb0ed75-39c9-4685-90fa-6c974f793a40)

**Comment by b3ck** [comment]
Normal Mode:
<img width="303" alt="Screen Shot 2024-08-03 at 3 00 24 PM" src="https://github.com/user-attachments/assets/997c6003-7dc5-417d-aa13-62fbae4729e8">

Boost Mode:
<img width="293" alt="Screen Shot 2024-08-03 at 3 00 33 PM" src="https://github.com/user-attachments/assets/e8d61979-7183-4e93-bcd1-0f252f31b39e">

**Comment by ginbot86** [comment]
Interesting... The Kraze variant uses the plain flame graphic for the boost mode.

**Comment by b3ck** [comment]
Yeah.. dunno.. must have been a approval type of thing with some community board in Canada.

**Comment by ginbot86** [comment]
I just went through some photos I took and the first Kraze HD Mega vape I found used the "tiger flames" boost mode graphic. Not sure when the change was made, or if it's consistent...

**Comment by ginbot86** [comment]
As mentioned in Issue 5, I have found the correlation between versions. Holding the side button for more than 10 seconds will cause a version number to display, and a V1 firmware uses the "tiger flames" graphic, and the V2 uses the plain flames.

**Comment by Rea-XEVION** [comment]
On the topic of refilling, years ago when I used to refill rechargeable dumb stick vapes, every single time they would eventually burn out the coil or put a charred hole in the wick even if I never let them run dry.
With the raz/kraze and even the v-play all having two coils, I have to wonder if a software change is possible to swap which coil is primary/secondary.  This would be useful in case someone gets a burned taste with the normal mode, but not with boost.

**Comment by ginbot86** [comment]
It would require some firmware reverse-engineering but I think it could be possible. This is speculation on my part but I imagine that the references to certain GPIOs could be find-and-replaced to swap the roles of the heater coils.

**Comment by Estacaco** [comment]
Just a note on Firmware versions while the Geek Bar has a completely different looking UI, it is running firmware version "AM.V3"

**Comment by martinbogo** [comment]
I just got a whole box of used RAZ Vape DC 25000 units.  My original plan was just to disassemble them and extract the screens and batteries for use in another project... but the coils in these units seems to be made **_extremely_** well and when I tested them I found they don't burn out easily ... in fact, they seem to be close to the quality of "non-disposable" coils.

So, I want to pivot to refurbishing them, drilling/milling a fill port on the side, and then giving them a second life.  Washing out the storage tank pads with clean 70% ethanol/H20 seems to be the way to go with these and they are easy to disassemble/reassemble from the bottom.

Is the only way to change the graphics, and perform a "reset" on the timer a chip-off solution?  Is the intermediate flash firmware via SWD possible on this platform?

**Comment by martinbogo** [comment]
> Just a note on Firmware versions while the Geek Bar has a completely different looking UI, it is running firmware version "AM.V3"

I did a disassembly of the Geek Bar Pulse X -- was pleasantly surprised to see it has a Cortex M0 and flash so it's in theory flashable and programmable, but I haven't yet managed to figure out how to SWD the thing.  

I'll open another thread on the fun I've had with the Geek Bar hardware...

**Comment by marylemons6** [comment]
It might feel exciting to visit free comic sites that offer endless reading options. But many of them hide dangers behind their [adventure content](https://xn--ij2bx6jk1s.store/), like malware, fake ads, or stolen material. These unsafe sites can harm your device or leak your personal details. To stay protected, always use official and secure comic platforms for reading.
